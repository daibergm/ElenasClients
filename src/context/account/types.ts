export enum ActionTypes {
  LOGIN_ATTEMPT = '@ACCOUNT_LOGIN_ATTEMPT',
  LOGIN_SUCCESS = '@ACCOUNT_LOGIN_SUCCESS',
  LOGIN_FAILURE = '@ACCOUNT_LOGIN_FAILURE',
  LOGOUT = '@ACCOUNT_LOGOUT',
}

export type Action = {
  type:
    | ActionTypes.LOGIN_ATTEMPT
    | ActionTypes.LOGIN_SUCCESS
    | ActionTypes.LOGIN_FAILURE
    | ActionTypes.LOGOUT;
  payload?: string;
};
