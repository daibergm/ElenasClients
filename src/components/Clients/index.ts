export { default as ClientsComponent } from './ClientsComponent';
