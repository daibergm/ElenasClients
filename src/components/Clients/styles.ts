import { StyleSheet } from 'react-native';

// @Assets
import { Colors } from '../../assets';

export default StyleSheet.create({
  container: {
    paddingHorizontal: 15,
    paddingVertical: 25,
  },
  emptyContainer: {
    flex: 1,
    justifyContent: 'center',
  },
  itemContainer: {
    paddingHorizontal: 15,
    paddingVertical: 20,
    backgroundColor: Colors.white,
    borderRadius: 5,
    marginBottom: 10,
    shadowColor: Colors.black,
    shadowOffset: {
      width: 0,
      height: 1,
    },
    shadowOpacity: 0.05,
    shadowRadius: 1,
    elevation: 1,
  },
});
