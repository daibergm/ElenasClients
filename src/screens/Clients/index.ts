export { default as ClientsScreen } from './ClientsScreen';
