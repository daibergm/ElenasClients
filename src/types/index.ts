export * from './routes';
export * from './user';
export * from './client';
