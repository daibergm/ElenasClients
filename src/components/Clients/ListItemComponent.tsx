import React from 'react';
import { TouchableOpacity, Text } from 'react-native';

// @Types
import { Client } from '../../types';

import styles from './styles';

interface Props {
  item: Client;
}

function ListItemComponent({ item }: Props) {
  return (
    <TouchableOpacity style={styles.itemContainer}>
      <Text>{`${item.firstName} ${item.lastName}`}</Text>
      <Text>{`Celular: ${item.cellphone}`}</Text>
    </TouchableOpacity>
  );
}

export default ListItemComponent;
