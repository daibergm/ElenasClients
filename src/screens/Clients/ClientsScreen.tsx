import React from 'react';
import { SafeAreaView, Text } from 'react-native';

// @Styles
import styles from './styles';

function ClientsScreen() {
  return (
    <SafeAreaView style={styles.container}>
      <Text>Lista de clientes</Text>
    </SafeAreaView>
  );
}

export default ClientsScreen;
