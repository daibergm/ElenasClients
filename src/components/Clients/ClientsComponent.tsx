import React from 'react';
import { FlatList, ListRenderItem } from 'react-native';

// @Styles
import styles from './styles';

// @Types
import { Client } from '../../types';

// @Components
import { EmptyList } from '../common';
import ListItemComponent from './ListItemComponent';

interface Props {
  data: Client[];
}

function ClientsComponent({ data }: Props) {
  const getKeyExtractor = (item: Client) => `${item.id}`;

  const renderItem: ListRenderItem<Client> = ({ item }) => (
    <ListItemComponent item={item} />
  );

  const renderEmptyComponent = () => {
    return (
      <EmptyList description="Lo sentimos, no pudimos encontrar ningÃºn cliente." />
    );
  };

  return (
    <FlatList
      style={styles.container}
      showsVerticalScrollIndicator={false}
      data={data}
      keyExtractor={getKeyExtractor}
      renderItem={renderItem}
      ListEmptyComponent={!data?.length ? renderEmptyComponent : null}
      contentContainerStyle={!data?.length && styles.emptyContainer}
    />
  );
}

export default ClientsComponent;
