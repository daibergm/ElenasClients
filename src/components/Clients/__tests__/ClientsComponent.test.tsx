import React from 'react';
import { render } from '@testing-library/react-native';

// @Components
import Clients from '../ClientsComponent';

// @Types
import { Client } from '../../../types';

describe('Clients test suits', () => {
  const data: Client[] = [
    {
      id: 1,
      firstName: 'John',
      lastName: 'Doe',
      cellphone: '123',
    },
  ];

  it('Render correctly', () => {
    const clients = render(<Clients data={data} />);
    expect(clients).toMatchSnapshot();
  });
});
