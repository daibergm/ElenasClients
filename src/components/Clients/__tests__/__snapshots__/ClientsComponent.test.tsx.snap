// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clients test suits Render correctly 1`] = `
Array [
  <View
    style={
      Object {
        "marginTop": 25,
        "paddingHorizontal": 15,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "borderRadius": 3,
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 3,
          },
          undefined,
          false,
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#00adee",
              "borderColor": "#2089dc",
              "borderRadius": 20,
              "borderWidth": 0,
              "flexDirection": "row",
              "height": 40,
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingVertical": 1,
                "textAlign": "center",
              }
            }
          >
            CREAR CLIENTE
          </Text>
        </View>
      </View>
    </View>
  </View>,
  <RCTScrollView
    ListEmptyComponent={null}
    contentContainerStyle={false}
    data={
      Array [
        Object {
          "cellphone": "123",
          "firstName": "John",
          "id": 1,
          "lastName": "Doe",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "paddingHorizontal": 15,
        "paddingVertical": 25,
      }
    }
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 5,
              "elevation": 1,
              "marginBottom": 10,
              "opacity": 1,
              "paddingHorizontal": 15,
              "paddingVertical": 20,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.05,
              "shadowRadius": 1,
            }
          }
        >
          <Text>
            John Doe
          </Text>
          <Text>
            Celular: 123
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>,
]
`;
